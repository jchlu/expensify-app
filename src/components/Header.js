import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = (props) => {
  return (
    <div className="header">
      <h1 className="header__title">{props.title}</h1>
      <p>
        |&nbsp;<NavLink className="header__link" activeClassName="header__link--active" to="/" exact={true}>Go Home</NavLink>&nbsp;
          |&nbsp;<NavLink className="header__link" activeClassName="header__link--active" to="/create">Add Expense</NavLink>&nbsp;
          |&nbsp;<NavLink className="header__link" activeClassName="header__link--active" to="/help">Help</NavLink>&nbsp;|
      </p>
    </div>
  )
}

Header.defaultProps = {
  title: 'Expensify'
}
export default Header
