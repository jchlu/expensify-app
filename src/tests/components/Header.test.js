import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from '../../components/Header'

test('Should render Header correctly', () => {
  const wrapper = shallow(<Header />)
  /**
   * Examples:
   * expect(wrapper.find('h1').length).toBe(1) // Check for only one h1 element
   * expect(wrapper.find('h1').text()).toBe('Expensify') // Check for element text
  */

  expect(toJson(wrapper)).toMatchSnapshot()
  // const renderer = new ReactShallowRenderer()
  // renderer.render(<Header />)
  // expect(renderer.getRenderOutput()).toMatchSnapshot()
})
