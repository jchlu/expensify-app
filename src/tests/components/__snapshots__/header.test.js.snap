// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<div
  className="header"
>
  <h1
    className="header__title"
  >
    Expensify
  </h1>
  <p>
    | 
    <NavLink
      activeClassName="header__link--active"
      ariaCurrent="true"
      className="header__link"
      exact={true}
      to="/"
    >
      Go Home
    </NavLink>
      | 
    <NavLink
      activeClassName="header__link--active"
      ariaCurrent="true"
      className="header__link"
      to="/create"
    >
      Add Expense
    </NavLink>
      | 
    <NavLink
      activeClassName="header__link--active"
      ariaCurrent="true"
      className="header__link"
      to="/help"
    >
      Help
    </NavLink>
     |
  </p>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
